ui/app/services/auth.js

```diff
@@ -296,21 +306,16 @@ export default Service.extend({
       resp.policies
     );
 
-    if (resp.renewable) {
-      Object.assign(data, this.calculateExpiration(resp));
```

Comment:
the `this.calculateExpiration` function is already set up to handle batch tokens returned by `/auth/[backend]/login` methods since [those APIs return a `lease_duration`](https://developer.hashicorp.com/vault/api-docs/auth/userpass#login), which is accounted for on line 220 above. 

as a result, i didn't see a reason why we wouldn't couldn't call `calculateExpiration` regardless of whether or not `resp.renewable` or not. 

so you'll notice i modified `this.calculateExpiration` to handle the case where `resp.type === 'batch'` (which happens when [logging in via a batch token directly](https://developer.hashicorp.com/vault/api-docs/auth/token#lookup-a-token-self)). this way `calculateExpiration` is used universally, regardless of what authentication method (token/userpass/ldap, batch token, service token etc).